Taskfile.yml

```diff
@@ -1,16 +1,223 @@
 version: '3'
 
-includes:
-  generator:
-    taskfile: ./hack/generator
-    dir: ./hack/generator/
-  generated:
-    taskfile: ./hack/generated
-    dir: ./hack/generated/
+vars:
+  GENERATOR_APP: k8sinfra-gen
+  GENERATOR_ROOT: ./hack/generator/
+
+  CONTROLLER_APP: k8sinfra-controller
+  CONTROLLER_ROOT: ./hack/generated/
+
+  # the 'api' dir contains generated code so we want to exclude it
+  # for go-fmt the root dir must be excluded (it recurses)
+  GENERATED_DIRS_TO_FMT_CMD: go list -f '{{"{{.Dir}}"}}' ./...  | grep -vE '/apis(/|$)|/generated$' | tr '\n' ' '
+  # for lint/test the root dir must be included
+  GENERATED_DIRS_TO_LINT_CMD: go list -f '{{"{{.Dir}}"}}' ./...  | grep -vE '/apis(/|$)' | tr '\n' ' '
 
 tasks:
+  default:
+    desc: Perform all fast local pre-commit tasks.
+    cmds: 
+    - task: generator:quick-checks
```

Comment:
Shouldn't the default target also do a build?

Reply:
It runs tests which should check all the code?

Reply:
Ahh, didn't realize the test target automatically did a build - I guess it must though? (go test contains go build?).

Sounds good.

Reply:
This now does a build anyway because the `generator:default/:ci` has to do a `generate-types` anyway (to prevent two concurrent `generate-types` happening in the `controller:default/:ci`… `go-task` should be fixing this soon I think)